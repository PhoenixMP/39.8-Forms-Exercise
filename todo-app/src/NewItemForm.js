import React, { useState } from "react";

/** Form for creating a new box to add to a list.
 *
 * Has state for the name/quantity of the item; on submission,
 * sends {color, width, height} to fn rec'd from parent.
 *
 */

const NewItemForm = ({ addItem }) => {
    const INITIAL_STATE = { name: "" };
    const [formData, setFormData] = useState(INITIAL_STATE);

    /** Send {name} to parent
     *    & clear form. */

    const handleSubmit = evt => {
        evt.preventDefault();
        addItem(formData);
        setFormData(INITIAL_STATE);
    };

    /** Update local state w/curr state of input elem */

    const handleChange = evt => {
        const { name, value } = evt.target;
        setFormData(fData => ({
            ...fData,
            [name]: value
        }));
    };

    /** render form */

    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="name">Todo List Item:</label>
            <input
                id="name"
                name="name"
                value={formData.name}
                onChange={handleChange}
            />
            <button>Add a new Item!</button>
        </form>
    );
};

export default NewItemForm;
