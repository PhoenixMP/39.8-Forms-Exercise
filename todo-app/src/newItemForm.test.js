import React from "react";
import { render } from "@testing-library/react";
import NewItemForm from "./NewItemForm";

it("renders without crashing", function () {
    render(<NewItemForm />);
});

it("matches snapshot", function () {
    const { asFragment } = render(<NewItemForm />);
    expect(asFragment()).toMatchSnapshot();
});
