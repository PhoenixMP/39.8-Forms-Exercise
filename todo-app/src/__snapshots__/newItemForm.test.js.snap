// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="name"
    >
      Todo List Item:
    </label>
    <input
      id="name"
      name="name"
      value=""
    />
    <button>
      Add a new Item!
    </button>
  </form>
</DocumentFragment>
`;
